src/main.js

```diff
@@ -28,15 +28,22 @@ Vue.config.silent = false;
 
 Vue.use(localStorage); // First, vm.$localStorage property
 
-// Google analytics
+Vue.use(gdpr);
+
+let enableAnalytics = false;
+try {
+  enableAnalytics = JSON.parse(window.localStorage.getItem('Gdpr.preferences', '{}')).choice?.statistics || false;
```

Comment:
Going deep in localStorage internals may be error prone.

You can use `beforeFirstHit` hook on option sent to VueAnalytics:

```Vue.use(VueAnalytics, {
  id: config.googleAnalyticsKey,
  disabled: true,
  beforeFirstHit() {
    if (Vue.prototype.$gdpr.gdprValue?.statistics) {
      Vue.prototype.$ga.enable();
    }
  },
```

https://github.com/MatteoGabriele/vue-analytics/blob/master/docs/when-google-analytics-is-loaded.md

(`ready()` is called **after** first GA request, so `beforeFirstHit` is the one to go)

Reply:
That was the way to go, thanks.